setup.py

```diff
@@ -4,6 +4,39 @@
 
 from setuptools import find_packages, setup
 
+REQUIRED_PYTHON = (3, 5)
+
+
+# NOTE: this check must precede anything that would break on Python 2.7
+# Fail with explicit error if trying to install Django on an incompatible
+# (too old) version of Python, and advise on how to avoid this.
+HAS_PYTHON = sys.version_info[:2]
+if HAS_PYTHON < REQUIRED_PYTHON:
+    sys.stderr.write("""
+
+===========
+FATAL ERROR
+===========
+
+This version of Django requires Python {}.{}, but you are trying to
+install it on Python {}.{}.
+
+This is probably because you are using an old version of pip, which
+does not understand the python_requires classifier. Make sure you
+have pip >= 9.0 and setuptools >= 24.2, then try again:
+
+    $ python -m pip install --upgrade pip setuptools
+    $ python -m pip install django
+
+This will install the latest version of Django which works on your
+version of Python. If you can't upgrade your pip (or Python),
+explicitly request an older version of Django:
+
+    $ python -m pip install "django<2"
+
+""".format(*(REQUIRED_PYTHON + HAS_PYTHON)))
```

Comment:
With unpacking generalizations in Python 3.5 you don't need to concatenate the tuples:
```python
.format(*REQUIRED_PYTHON, *HAS_PYTHON)
```

Reply:
This needs to remain compatible with Python 2.7 as discussed and commented.

Reply:
Oh yeah... That was silly of me :blush: